### PR TITLE
[SpringBoot 2.3] R2DBC

### PR DESCRIPTION
**Pull request by murdos**
-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by murdos** [comment]
@mraible : Thanks, but wait a bit, it's not ready yet 😁

File: generators/server/templates/src/main/java/package/repository/UserRepository.java.ejs [hunk 1]
```diff
@@ -173,7 +173,7 @@ public interface UserRepository extends <% if (databaseType === 'sql') { %>JpaRe
     <%_ if (databaseType === 'neo4j') { _%>
     <% if (!reactive) { %>// See https://github.com/neo4j/sdn-rx/issues/51<% } %>
     <% if (reactive) { %>Flux<% } else { %>List<% } %><<%= asEntity('User') %>> findAll();
-    
+
```
Comment (cbornet) [inline generators/server/templates/src/main/java/package/repository/UserRepository.java.ejs:176]: I think query derivation is supported with this release. So the `@Query` annotations can be removed.
And `Pageable` should also supported in query derivation.

Comment (murdos) [reply generators/server/templates/src/main/java/package/repository/UserRepository.java.ejs:176]: It is. But I would like to fix tests before doing some cleanup: I think I've found a regression in Spring R2DBC, I'm trying to recreate a minimal project to illustrate the issue and report it.

**Comment by murdos** [comment]
This PR is currently blocked by this issue: https://github.com/spring-projects/spring-data-r2dbc/issues/369

**Pull request merged**